glusterfs.spec.in

```diff
@@ -334,6 +338,7 @@ and client framework.
 
 %package cli
 Summary:          GlusterFS CLI
+BuildRequires:    pkgconfig(bash-completion)
```

Comment:
I'm not sure if we really need to force this dependency.

Reply:
I am not sure either.

For systems where bash-completion and its pkgconf are easily available, I guess it should: to be able to install the script the recommended way so that it is loaded dynamically. For systems where it might not be the case, the default in `/etc/bash-completion.d` is fine anyway, so the requirement is not needed.

Could a condition be added around the BuildRequires?

Reply:
I don't know if it can be done conditionally.

The problem I see is that if the requirement cannot be satisfied for some reason, the build will fail, when it shouldn't. Not sure what's the right way to manage this.

Reply:
I agree that the package build should not break on that. If it is not available the default compatible path is enough.

The pkg-config definitions were added in bash-completion in version 1.90 [[1]]. Supported distros seem to have 2.x versions. (The devrpms-el7 job build it ok with the new path.)

[1]: https://github.com/scop/bash-completion/commit/90b67bc976dccc655f7ba89714b6c1e5b1e5ddc5

There are spec macros to check on distribution and distribution version. There are some used in the spec file (eg for systemd).
Maybe a condition for RHEL would be needed, but I don't know how to check that.

Reply:
As this is 'only' a `BuildRequires`, it will only be needed to be installed while building RPMs. If for some reason it is not available on a distribution, it should be simple for a packager to provide a workaround for that particular distribution.

Reply:
If we determine distros or any other parameters where we have the required package available for sure, then we can add a conditional similar to: https://github.com/gluster/glusterfs/blob/devel/glusterfs.spec.in#L291

Also adding a comment explaining that the default in /etc/bash-completion.d would work incase the build requires of this package cannot be met due to its unavailability in any system would help a packager to modify this requirement

Reply:
Thank you for your review.

I added a comment and a condition to exclude old RHEL versions as an example. Most rpm distributions that are supported today have recent enough version for bash-completion, so the idea is to exclude some old distribution versions instead of specifically allowing most of the current ones. I hope it is ok.